sshfs.c

```diff
@@ -1824,7 +1824,9 @@ static void *sshfs_init(void)
 #if FUSE_VERSION >= 26
 	/* Readahead should be done by kernel or sshfs but not both */
 	if (conn->async_read)
-		sshfs.sync_read = 1;
+		// Force async even if kernel claims its doing async.
+		// MacOS, see https://github.com/osxfuse/sshfs/issues/57
+		sshfs.sync_read = 0;
```

Comment:
Just FYI, I cloned, compiled and tried this and got much worst performances... there might be other factors at play such as my fuse version.

Reply:
An update about the speed issues.  
I have tested this update :
https://github.com/osxfuse/sshfs/pull/58

It is working very well. 
Transferred 216 GB from Norway to Switzerland.

sshfs for OSX version 2.10.0 
force ssh-fs>sync_read = 0 

The sshfs gives me a speed of 7,2 Gigabyte / hour = 160 Megabit / second = 20 MB / s
sshfs test on a 250 Mbps line. This is good results. The transfer speed is good. 

Running rsync from synology NAS - Norway to Switzerland, will only give me 120 Megabit / second = 15 MB / s
The rsync test was on a  300 Mbps internet line

So the speed issue can be solved.
Work around is to force sshfs to do async -reads even if the kernel claims it's doing it.
There might be more elegant ways of solving this, but it is definitely working.